src/components/Contacts/ContactFilters/ContactFilters.tsx

```diff
@@ -59,14 +86,18 @@ export const ContactFilters: React.FC<Props & BoxProps> = ({
   });
 
   const [selectedGroup, showGroup] = useState<FilterGroup>();
-  const [selectedFilters, setSelectedFilters] = useState<{
-    [name: string]: boolean | string | Array<string>;
-  }>({});
+  const [selectedFilters, setSelectedFilters] = useState<ContactFilterSetInput>(
+    {},
+  );
   const [showAll, setShowAll] = useState(false);
 
+  useEffect(() => {
+    onSelectedFiltersChanged(selectedFilters);
+  }, [selectedFilters]);
```

Comment:
This could be done in `updateSelectedFilter` too. Which do you think is better? I'm ok with this, just usually try and justify extra `useEffect` hooks.

Reply:
Yeah that sounds good.  I know what you mean, adding more hooks in general could bog down the screen with extra rerenders.  `useEffect` made sense to me because we're mostly responding to changes in the `selectedFilters` state, but calling this callback directly after setting the state is good because it reduces the need for hooks.